app/Services/Entity/AbilityService.php

```diff
@@ -51,17 +51,27 @@ public function abilities(): array
                 // entity
                 'ability.entity', 'ability.entity.image', 'ability.entity.attributes',
                 // parent
-                'ability.ability', 'ability.ability.entity'
+                'ability.ability', 'ability.ability.entity', 'ability.ability.tags',
             ])
             ->join('abilities as a', 'a.id', 'entity_abilities.ability_id')
             ->defaultOrder()
             ->get();
         /** @var EntityAbility $ability */
-        foreach ($abilities as $ability) {
+        foreach ($abilities as $key => $ability) {
             // Can't read the ability? skip
             if (empty($ability->ability) || empty($ability->ability->entity)) {
                 continue;
             }
+            $class = '';
```

Comment:
Why not have this whole block in the `format()` method, since that's also where the ability is transformed into json?